tools/coredump/testdata/amd64/ruby-3.5.0-loop.json

```diff
@@ -4,27 +4,26 @@
     {
       "lwp": 1844728,
       "frames": [
-        "is_prime+0 in /app/loop.rb:10",
-        "is_prime+0 in /app/loop.rb:10",
-        "sum_of_primes+0 in /app/loop.rb:20",
-        "<main>+0 in /app/loop.rb:30",
+        "Object#is_prime+0 in /app/loop.rb:0",
         "libruby.so.3.5.0+0x3399b0",
         "libruby.so.3.5.0+0x33e9d7",
         "libruby.so.3.5.0+0x23e175",
         "libruby.so.3.5.0+0x31e2ea",
         "libruby.so.3.5.0+0x325240",
         "libruby.so.3.5.0+0x333b86",
-        "<main>+0 in /app/loop.rb:29",
-        "loop+0 in <internal:kernel>:168",
-        "<main>+0 in /app/loop.rb:28",
+        "Range#each+0 in <cfunc>:0",
+        "Object#is_prime+0 in /app/loop.rb:10",
+        "Object#sum_of_primes+0 in /app/loop.rb:20",
+        "block (2 levels) in <main>+0 in /app/loop.rb:30",
         "libruby.so.3.5.0+0x339d65",
         "libruby.so.3.5.0+0x33e9d7",
         "libruby.so.3.5.0+0x23e3bd",
         "libruby.so.3.5.0+0x31ce80",
         "libruby.so.3.5.0+0x325240",
         "libruby.so.3.5.0+0x333b86",
-        "<main>+0 in /app/loop.rb:29",
-        "loop+0 in <internal:kernel>:168",
+        "Range#each+0 in <cfunc>:0",
+        "block in <main>+0 in /app/loop.rb:29",
```

Comment:
Notice that a lot of these now have more context because we have conditional logic picking between "label", "base_label", and "method name"  from the iseq, to match ruby's own logic for this.